hack/release-helper.sh

```diff
@@ -0,0 +1,257 @@
+    echo "          Note: Takes no args."
+    echo "  update  Update the operator repo with new tags. Modifies local"
+    echo "          files but does not commit or push them."
+    echo "          Note: All args are required"
+    echo
+    echo "  ARGS"
+    echo "  -o  The operator tag for your release"
+    echo "      Example: v0.9.0"
+    echo "  -e  The enclave-cc tag for your release"
+    echo "      Example: v0.9.1"
+    echo "  -k  The kata tag for your release (n.b. no 'v' prefix)"
+    echo "      Example: 3.6.0"
+    echo
+    echo "Example usage:"
+    echo "    $0 check"
+    echo "    $0 update -o v0.9.0 -e v0.9.1 -k 3.6.0"
+    echo
+    exit 1
+}
+
+
+function bail_if_dirty_repo() {
+    git diff --quiet ||
+          (echo
+           echo "Error: Must use a clean repo to use this script."
+           echo "       Stash, commit, branch, or use a fresh clone, etc."
+           echo
+           exit 1)
+}
+
+
+function parse_args() {
+    # pull the ACTION from the args list
+    if [ $# = 0 ]; then
+        usage_and_exit
+    fi
+    action=$1
+    shift
+
+    # parse the o, e, and k args
+    while getopts ":o:e:k:" opt; do
+        case "${opt}" in
+            o)
+                operator_tag=${OPTARG} ;;
+            e)
+                enclave_cc_tag=${OPTARG} ;;
+            k)
+                kata_tag=${OPTARG} ;;
+            *)
+                usage_and_exit ;;
+        esac
+    done
+
+    # minimal checking: ensure all args are provided when using 'update'
+    if [ ${action} = "update" ]; then
+        if [[ ! -v operator_tag ]] || [[ ! -v enclave_cc_tag ]] || [[ ! -v kata_tag ]]; then
+            echo "Error: Missing ARGS"
+            usage_and_exit
+        fi
+    fi
+}
+
+
+#
+# Make sure that the pre-reqs container ships the same version of the nydus
+# snapshotter that we use in the kata ci. Update the Makefile if needed.
+#
+function check_nydus() {
+    echo "Checking that the operator's nydus version matches the one used in kata"
+
+    kata_versions="https://raw.githubusercontent.com/kata-containers/kata-containers/main/versions.yaml"
+    kata_versions_dest="versions.yaml.$$"
+
+    # pull the versions.yaml that kata uses
+    wget --quiet "${kata_versions}" -O "${kata_versions_dest}"
+
+    # multi-line grep to find the nydus-snapshotter version kata's version.yaml
+    # then awk/tail/tr and string manipulation to sand it down.
+    nydus_version_kata="$(grep -zoP "nydus-snapshotter.*\n.*description.*\n.*url.*\n.*version.*" ${kata_versions_dest} | awk '{print $2}' | tail -n1 | tr -d '\0')"
+    nydus_version_kata=${nydus_version_kata//\"} # strip double quotes
+
+    # delete temporary kata versions.yaml that we downloaded
+    rm "${kata_versions_dest}"
+
+    # grab the nydus version in the operator
+    prereqs_makefile="install/pre-install-payload/Makefile"
+    nydus_version_prereqs="$(grep 'NYDUS_SNAPSHOTTER_VERSION = ' $prereqs_makefile | awk '{ print $3}')"
```

Comment:
Nit: for a more precise match use `^NYDUS_SNAPSHOTTER_VERSION = `

Reply:
This got resolved before I saw it... but your comment is still valid, no? I'm making your suggested change.